api/lib/domain/models/user-campaign-results/StageCollection.js

```diff
@@ -43,6 +48,10 @@ class StageCollection {
       reachedStage: reachedStageIndex + 1,
     };
   }
+
+  getReachedStageForCompetence(validatedSkillCount, masteryPercentage) {
+    return this._getReachedStageIndex(validatedSkillCount, masteryPercentage) + 1;
+  }
```

Comment:
Renommer pour être plus explicite sur ce qui est retourné par la méthode ?

```suggestion
  getReachedStageIndexForCompetence(validatedSkillCount, masteryPercentage) {
    return this._getReachedStageIndex(validatedSkillCount, masteryPercentage) + 1;
  }
```

Reply:
Ici c'est bien le reachedStageForCompetence qu'on aimerait retourner :) Le nommage me paraît correct

Reply:
Je me suis fait avoir aussi, je comprends l'incompréhension de Geoffroy. Ici, on retourne un nombre, alors que la méthode juste au dessus `getReachedStage` retourne un objet. 

Je me demande pourquoi on ne retourne pas l'objet ? 

Si je recap : 
Un reachedStage actuellement c'est un objet qui a aussi une clé reachedStage 🤯 : 
```
{
        id: null,
        totalStage: null,
        reachedStage: null,
        prescriberTitle: null,
        prescriberDescription: null,
}
```
Et maintenant un `reachedStageForCompetence` c'est un nombre 
C'est bien ça ? Pourquoi ce changement ?

Reply:
Oui c'est ça ! Aujourd'hui, il  y a 1 termes métier pour 2 concepts : l'object ReachedStage qui contient lui même un reachedStage qui est un nombre. Je trouve ça également confusant et cela mériterait peut-être une plus grosse refacto prochainement (à discuter avec l'équipe qui gère les target profile et les palliers?)

Concernant notre besoin, on a seulement besoin d'un nombre reachedStage pour la compétence :)

Reply:
Ca marche, l'explication mérite sa place dans la description de PR

Reply:
Oui je ne sais pas si ça "mérite" mais effectivement je peux l'ajouter :)